lib/server-middleware.js

```diff
@@ -16,19 +30,40 @@ module.exports = options => async (req, res, next) => {
 
   const handler = new Handler({ req, res, next, options })
 
+  const existingToken = extractToken(req)
+  let invalidToken = false
+
+  // when an existing token exists try to authenticate the session
+  if (existingToken) {
+    await handler.saveData({ accessToken: existingToken })
+
+    // If we have a user, the token was valid
+    if (!isEmptyObject(req.user)) {
```

Comment:
this logic seems to rely on some magic inside the handler (i.e. run `saveData` and then check `req.user`)

This whole block should probably move inside the handler, so we can just run `handler.checkHeaderForToken()` or something.

This should probably also be after the other `if` statements, as if the route is `login`, `callback` or `logout` we probably don't care about a token in the header.

In which case we can just do `if (handler.checkHeaderForToken()) return` and that can handle the entire thing including logging out if not valid?

Reply:
Sure we can move it later in the sequence, and inside the handler.

Reply:
`fetchUser` is from external resources and doesn't return errors just empty objects (which won't fail inside the current library implementation). So checking if the user exists is the "easiest" way of getting this working.

We can change the logic to make it handle errors more appropriately, then we should be able to use errors responses rather than check if the suer exists.

Happy to incorporate this into this PR, if wanted.

Reply:
@samtgarson 
Now we don't check for user and i've removed that piece.

It just loads as it did previously.